src/PSR7/Validators/PathValidator.php

```diff
@@ -42,18 +41,14 @@ public function validate(OperationAddress $addr, MessageInterface $message) : vo
      */
     private function validateRequest(OperationAddress $addr, RequestInterface $message) : void
     {
-        $specs = $this->finder->findPathSpecs($addr);
-
+        $validator        = new ArrayValidator($this->finder->findPathSpecs($addr));
         $path             = $message->getUri()->getPath();
         $pathParsedParams = $addr->parseParams($path); // ['id'=>12]
-        $validator        = new SchemaValidator($this->detectValidationStrategy($message));
-
-        foreach ($pathParsedParams as $name => $value) {
-            try {
-                $validator->validate($value, $specs[$name]->schema);
-            } catch (SchemaMismatch $e) {
-                throw InvalidPath::becauseValueDoesNotMatchSchema($name, (string) $value, $addr, $e);
-            }
-        }
+
+        try {
+            $validator->validateArray($addr, $pathParsedParams, $this->detectValidationStrategy($message));
+        } catch (InvalidParameter $e) {
+            throw InvalidPath::becauseValueDoesNotMatchSchema($e->name(), $e->value(), $addr, $e->getPrevious());
+        } // RequiredParameterMissing will not be thrown, all parameters are checking in parseParams
```

Comment:
Well in this case I suggest rethrow to a `\LogicException`. This will never happen, but if unfortunatelly do - lib users could notify us to fix problem somewhere

Reply:
Without it any stat analyzer will think that this method will be able to throw `RequiredParameterMissing`

Reply:
Good point, fixed it